src/services/safe_border.py

```diff
@@ -135,25 +138,19 @@ def _get_earliest_slashed_epoch_among_incomplete_slashings(self) -> EpochNumber
         # Here we filter not by exit_epoch but by withdrawable_epoch because exited operators can still be slashed.
         # See more here https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#helpers
         # at `get_eligible_validator_indices` method.
-        validators_slashed_non_withdrawable = filter_non_withdrawable_validators(validators_slashed,
-                                                                                 self.blockstamp.ref_epoch)
+        validators_slashed_non_withdrawable = filter_non_withdrawable_validators(validators_slashed, self.blockstamp.ref_epoch)
 
         if not validators_slashed_non_withdrawable:
             return None
 
-        validators_with_earliest_exit_epoch = self._filter_validators_with_earliest_exit_epoch(
-            validators_slashed_non_withdrawable)
-
         earliest_predicted_epoch = None
-
-        for validator in validators_with_earliest_exit_epoch:
+        for validator in validators_slashed_non_withdrawable:
             predicted_epoch = self._predict_earliest_slashed_epoch(validator)
 
             if not predicted_epoch:
-                return self._find_earliest_slashed_epoch_rounded_to_frame(validators_with_earliest_exit_epoch)
+                return self._find_earliest_slashed_epoch_rounded_to_frame(validators_slashed_non_withdrawable)
 
-            if not earliest_predicted_epoch or earliest_predicted_epoch > predicted_epoch:
-                earliest_predicted_epoch = predicted_epoch
+            earliest_predicted_epoch = min(earliest_predicted_epoch or EpochNumber(sys.maxsize), predicted_epoch)
```

Comment:
I don't think we should use sys.maxsize here.
Pls, check what type is epoch number and setup it's value.
Or I'd better prefer old expression